packages/ec-core/src/operator/composable/repeat_with.rs

```diff
@@ -82,9 +77,9 @@ mod tests {
     fn deterministic() {
         const LENGTH: usize = 5;
         let desired_value = 7;
-        let mut rng = thread_rng();
+        let mut rng = rng();
         let repeater: RepeatWith<AddOne, LENGTH> = RepeatWith::new(AddOne);
-        let result = repeater.apply(desired_value, &mut rng).unwrap();
+        let Ok(result) = repeater.apply(desired_value, &mut rng);
```

Comment:
It can't panic, that's the thing - the compiler verified at compile time that the error variant Is uninhabited (i.e never constructed, a enum without variants, a type you can't actually construct) else the compiler wouldn't let you write this. Think of it as the compiler verifying that the type is `Result<T,, Infallible>`

Reply:
Think of it like the compiler narrowing the type to

```rust
enum Result<T,E> {
    Ok(T)
}
```

Reply:
That's super cool, and wasn't not obvious to me. So a comment is still probably a good idea, but that's really nifty.